code/modules/antagonists/malf_ai/datum_malf_ai.dm

```diff
@@ -0,0 +1,206 @@
+	return ..()
+
+/// Callback for COMSIG_MOVABLE_HEAR which highlights syndicate code phrases in chat.
+/datum/antagonist/malf_ai/proc/handle_hearing(datum/source, list/hearing_args)
+	SIGNAL_HANDLER
+
+	var/message = hearing_args[HEARING_RAW_MESSAGE]
+	message = GLOB.syndicate_code_phrase_regex.Replace(message, "<span class='blue'>$1</span>")
+	message = GLOB.syndicate_code_response_regex.Replace(message, "<span class='red'>$1</span>")
+	hearing_args[HEARING_RAW_MESSAGE] = message
+
+/// Adds an objective datum to the objective list.
+/datum/antagonist/malf_ai/proc/add_objective(datum/objective/O)
+	objectives += O
+
+/// Removes an objective datum from the objective list.
+/datum/antagonist/malf_ai/proc/remove_objective(datum/objective/O)
+	objectives -= O
+
+/// Generates a complete set of malf AI objectives up to the traitor objective limit.
+/datum/antagonist/malf_ai/proc/forge_ai_objectives()
+	objectives.Cut()
+
+	if(prob(PROB_SPECIAL))
+		forge_special_objective()
+
+	var/objective_limit = CONFIG_GET(number/traitor_objectives_amount)
+	var/objective_count = length(objectives)
+
+	for(var/i in objective_count to objective_limit)
+		var/datum/objective/assassinate/kill_objective = new
+		kill_objective.owner = owner
+		kill_objective.find_target()
+		add_objective(kill_objective)
+
+	var/datum/objective/survive/malf/dont_die_objective = new
+	dont_die_objective.owner = owner
+	add_objective(dont_die_objective)
+
+/// Generates a special objective and adds it to the objective list.
+/datum/antagonist/malf_ai/proc/forge_special_objective()
+	var/special_pick = rand(1,4)
+	switch(special_pick)
+		if(1)
+			var/datum/objective/block/block_objective = new
+			block_objective.owner = owner
+			add_objective(block_objective)
+		if(2)
+			var/datum/objective/purge/purge_objective = new
+			purge_objective.owner = owner
+			add_objective(purge_objective)
+		if(3)
+			var/datum/objective/robot_army/robot_objective = new
+			robot_objective.owner = owner
+			add_objective(robot_objective)
+		if(4) //Protect and strand a target
+			var/datum/objective/protect/yandere_one = new
+			yandere_one.owner = owner
+			add_objective(yandere_one)
+			yandere_one.find_target()
+			var/datum/objective/maroon/yandere_two = new
+			yandere_two.owner = owner
+			yandere_two.target = yandere_one.target
+			yandere_two.update_explanation_text() // normally called in find_target()
+			add_objective(yandere_two)
+
+/datum/antagonist/malf_ai/greet()
+	to_chat(owner.current, "<span class='alertsyndie'>You are the [owner.special_role].</span>")
+	owner.announce_objectives()
+	if(should_give_codewords)
+		give_codewords()
+
+/datum/antagonist/malf_ai/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+
+	var/mob/living/silicon/ai/datum_owner = mob_override || owner.current
+	add_antag_hud(antag_hud_type, antag_hud_name, datum_owner)
+
+	if(istype(datum_owner))
+		datum_owner.hack_software = TRUE
+
+	RegisterSignal(datum_owner, COMSIG_MOVABLE_HEAR, .proc/handle_hearing)
```

Comment:
This doesn't make sense, it's going to try to call `handle_hearing` on the `/mob/living/silicon/ai` itself. Have you tested this? Is this what the traitor code is as well?

Reply:
This was universal in the old traitor code and applied to both traitors and AIs.

![image](https://user-images.githubusercontent.com/24975989/118057895-d1b09e00-b384-11eb-8e26-300650b37cf6.png)

![image](https://user-images.githubusercontent.com/24975989/118057874-c2c9eb80-b384-11eb-9b64-382cd75dae42.png)

Reply:
In testing, it works just fine as-is
![image](https://user-images.githubusercontent.com/24975989/118060436-b21c7400-b38a-11eb-8853-1c640d107a7b.png)

Reply:
Right. Componentised. Component takes a regex and a span. Component is dupe compatible and holds a reference to the antag datum which put it on the parent. This allows the antag datum to only delete the components that it added to the parent.

As Malf AI.
![image](https://user-images.githubusercontent.com/24975989/118066990-953a6d80-b397-11eb-9fe6-74308d2360d1.png)
![image](https://user-images.githubusercontent.com/24975989/118067020-a1bec600-b397-11eb-95ff-d7d8d2d7d5de.png)

With malf AI removed
![image](https://user-images.githubusercontent.com/24975989/118067126-db8fcc80-b397-11eb-8f54-c12ca48eb1a6.png)
![image](https://user-images.githubusercontent.com/24975989/118067147-e6e2f800-b397-11eb-82e8-aeafe133af5d.png)